lib/router/router.js

```diff
@@ -1,14 +1,14 @@
 /* global __NEXT_DATA__ */
 
-import { parse, format } from 'url'
+import { parse, format, resolve as resolveURL } from 'url'
 import evalScript from '../eval-script'
 import shallowEquals from '../shallow-equals'
 import { EventEmitter } from 'events'
 import { reloadIfPrefetched } from '../prefetch'
 import { loadGetInitialProps } from '../utils'
 
 export default class Router extends EventEmitter {
-  constructor (pathname, query, { Component, ErrorComponent, err } = {}) {
+  constructor (pathname, query, { Component, ErrorComponent, err, assetPrefix } = {}) {
```

Comment:
Cool, OK.

FWIW, the only usage of `global.__NEXT_DATA__` in this file is for the `buildId` property. We'll be able to remove the usage of the global in this PR =)